evm/utils/datatypes.py

```diff
@@ -9,36 +9,40 @@
 )
 
 
-def _is_local_prop(prop):
+from typing import Any, Dict, Tuple, Iterator, List
+
+
+def _is_local_prop(prop: str) -> bool:
     return len(prop.split('.')) == 1
 
 
-def _extract_top_level_key(prop):
+def _extract_top_level_key(prop: str) -> str:
     left, _, _ = prop.partition('.')
     return left
 
 
-def _extract_tail_key(prop):
+def _extract_tail_key(prop: str) -> str:
     _, _, right = prop.partition('.')
     return right
 
 
 @to_dict
-def _get_local_overrides(overrides):
+def _get_local_overrides(overrides: Dict[str, Any]) -> Iterator[Tuple[str, int]]:
     for prop, value in overrides.items():
         if _is_local_prop(prop):
             yield prop, value
 
 
 @to_dict
-def _get_sub_overrides(overrides):
+def _get_sub_overrides(overrides: Dict[str, Any]) -> Iterator[Tuple[str, Any]]:
```

Comment:
I don't think we could use something other than `Any` on the return value here, right? Usually we should avoid `Any` in return values as that makes it impossible for mypy to perform any checks on code that uses the values returned here

Reply:
>Usually we should avoid Any in return values as that makes it impossible for mypy to perform any checks on code that uses the values returned here

Yes, fully agree here. We probably can tighten that up. I admit, This was me being too lazy to dig deep into how this code is being used. I'm gonna give it a shot ;)

Reply:
Gave it another try but failed to figure out how I could tighten the bounds of `Any` here. Basically because I'm not sure what to expect here but my intuition says this could really be anything ;)

In general, I believe we'll probably have to iterate over some things and fine tune over time as more and more of the rest of the code base gets typed. As the overall coverage grows it should become easier to find the right types.